src/components/FeatureViews/Courses/AttendanceTable.js

```diff
@@ -0,0 +1,437 @@
+import IconButton from "@material-ui/core/IconButton";
+import ButtonGroup from "@material-ui/core/ButtonGroup";
+import Checkbox from "@material-ui/core/Checkbox";
+import { Add, Check } from "@material-ui/icons";
+import Loading from "../../OmouComponents/Loading";
+import {
+  StudentFilterOrSortDropdown,
+  SessionDropdownButton,
+} from "./AttendanceButtons";
+import { buttonThemeBlue } from "../../../theme/muiTheme";
+import { fullName } from "../../../utils";
+
+const useStyles = makeStyles((theme) => ({
+  table: {
+    maxWidth: "75.2vw",
+    overflowY: "auto",
+    [theme.breakpoints.down("lg")]: {
+      maxWidth: "69vw",
+    },
+  },
+  tableCell: {
+    color: "black",
+    whiteSpace: "nowrap",
+    fontWeight: 500,
+  },
+  checkbox: {
+    borderRadius: 2,
+    backgroundColor: buttonThemeBlue,
+    color: "white",
+    width: "1em",
+    height: "1em",
+    marginLeft: ".5em",
+    "&:hover": {
+      backgroundColor: buttonThemeBlue,
+    },
+  },
+  checkbox2: {
+    "&:hover": {
+      backgroundColor: "rgba(39, 143, 195, 1) !important",
+    },
+  },
+  buttonGroupStyle: {
+    border: 'none',
+    padding: 0,
+    minWidth: "1.5em",
+    minHeight: "1.5em"
+  },
+  buttonStatusStyle: {
+    maxWidth: "5.715em",
+    width: "100%",
+    maxHeight: "2em",
+    height: "100%",
+    textTransform: "none"
+  }
+}));
+
+// If not using studentId delete it
+function createStudentAttendanceRow(studentName, attendanceList, studentId) {
+  return { studentName, attendanceList, studentId };
+}
+
+export const GET_ATTENDANCE = gql`
+  query getAttendance($courseId: ID!) {
+    __typename
+    attendances(courseId: $courseId) {
+      id
+      status
+      session {
+        startDatetime
+        id
+      }
+      enrollment {
+        student {
+          user {
+            lastName
+            id
+            firstName
+          }
+        }
+      }
+    }
+    sessions(courseId: $courseId) {
+      id
+      startDatetime
+    }
+    enrollments(courseId: $courseId) {
+      student {
+        user {
+          firstName
+          id
+          lastName
+        }
+      }
+    }
+  }
+`;
+
+export const UPDATE_ATTENDANCE_STATUS = gql`
+  mutation UpdateAttendanceStatus(
+    $attendanceId: ID!
+    $status: AttendanceStatusEnum!
+  ) {
+    __typename
+    updateAttendance(id: $attendanceId, status: $status) {
+      attendance {
+        updatedAt
+      }
+    }
+  }
+`;
+
+const AttendanceTable = ({ setIsEditing }) => {
+  const { id } = useParams();
+  const classes = useStyles();
+  const [attendanceEditStates, setAttendanceEditStates] = useState();
+  const [isCheckBoxEditing, setIsCheckBoxEditing] = useState();
+  const [courseAttendanceMatrix, setCourseAttendanceMatrix] = useState([]);
+  const [sortByAlphabet, setSortByAlphabet] = useState("");
+  const [attendanceBySession, setAttendanceBySession] = useState({});
+
+  useEffect(() => {
+    if(courseAttendanceMatrix.length > 0) {
```

Comment:
write a comment at the top describing 1) what are you doing here (setting the sessions that are done editing to a "done" edit state 2) why are you doing this